nixos/modules/hardware/tiltfive.nix

```diff
@@ -0,0 +1,93 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
```

Comment:
```suggestion
```

Reply:
There's a bunch of `mkXXX` calls in there. Why is this `with lib;` a problem? Should it be nested somewhere deeper? Should I use `lib.mkXXX` explicitly?

Reply:
`with` is an anti-pattern too ingrained in old code. Indeed I am cleaning up this mess, and it is hard as hell.

> Why is this with lib; a problem? 

1. `with` works in a similar manner of `#include <>` from C language, bringing to the scope of your code a whole bunch of things you are not using. This is not so harmful in Nix because it uses lazy evaluation, however it can be a pain for static analysis, and can be worse if you accidentally invokes a functionality from it.
2. `with` works in unexpected ways when used more than once.
3. its scoping rules are obscure: https://github.com/NixOS/nix/issues/490

>  Should I use lib.mkXXX explicitly?

Yes.

Another less verbose alternative is to substitute from `with lib;` to `inherit (lib) mkXXX mkXXY mkXXZ . . .;`

(Usually @SuperSandro2000 prefers the explicit `lib.` usage, but I have no strong opinions between the options above.)

Reply:
> Another less verbose alternative is to substitute from `with lib;` to `inherit (lib) mkXXX mkXXY mkXXZ . . .;`

where you need to maintain a list of used functions which is also not that great.